src/output.rs

```diff
@@ -55,27 +115,38 @@ impl<'a, W: Write> Printer<'a, W> {
             }
         }
 
-        if let Some(ref format) = self.config.format {
-            self.print_entry_format(entry, format)?;
-        } else if self.config.yaml {
-            self.print_entry_detail(DetailFormat::Yaml, entry)?;
-        } else if self.config.json {
-            self.print_entry_detail(DetailFormat::Json, entry)?;
-        } else if let Some(ref ls_colors) = self.config.ls_colors {
-            self.print_entry_colorized(entry, ls_colors)?;
-        } else {
-            self.print_entry_uncolorized(entry)?;
+        match (
+            &self.config.format,
+            &self.config.output,
+            &self.config.ls_colors,
+        ) {
+            (Some(template), _, _) => self.print_entry_format(entry, template)?,
+            (None, OutputFormat::Json, _) => self.print_entry_detail(OutputFormat::Json, entry)?,
+            (None, OutputFormat::Yaml, _) => self.print_entry_detail(OutputFormat::Yaml, entry)?,
+            (None, OutputFormat::Ndjson, _) => {
+                self.print_entry_detail(OutputFormat::Ndjson, entry)?
+            }
+            (None, OutputFormat::Plain, Some(ls_colors)) => {
+                self.print_entry_colorized(entry, ls_colors)?
+            }
+            (None, OutputFormat::Plain, None) => self.print_entry_uncolorized(entry)?,
         };
 
         if has_hyperlink {
             write!(self.stdout, "\x1B]8;;\x1B\\")?;
         }
 
         self.started = true;
+
+        if matches!(
+            self.config.output,
+            OutputFormat::Json | OutputFormat::Ndjson
```

Comment:
```suggestion
            OutputFormat::Json 
```
I think we should allow nul output for `Ndjson`. Although then it is really nul separated instead of newline separated.